src/app/pages/SingleEntry/SingleEntry.tsx

```diff
@@ -1,27 +1,54 @@
 import styles from './SingleEntry.module.css';
 import Navigation from '../../components/Navigation/Navigation';
 import Button from '../../components/Button/Button';
+import { useParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
 
 type SingleEntryProps = {
-  headerTitle: string;
-  src: string;
-  entryTitle: string;
+  imageUrl: string;
+  date: string;
+  title: string;
   text: string;
 };
 
-function SingleEntry({
-  headerTitle,
-  src,
-  entryTitle,
-  text,
-}: SingleEntryProps): JSX.Element {
+function SingleEntry(): JSX.Element {
+  const { date } = useParams();
+  const [entry, setEntry] = useState<SingleEntryProps | null>();
+
+  const getEntries = async () => {
+    const response = await fetch(`/api/entries/${date}/`);
+    const entry = await response.json();
+    setEntry(entry);
+  };
+
+  useEffect(() => {
+    getEntries();
+  }, []);
+
+  const getDate = (dateString: string) => {
+    const dateObject = new Date(dateString);
+    const options: Intl.DateTimeFormatOptions = {
+      month: 'long',
+      day: 'numeric',
+      year: 'numeric',
+    };
+    const entryDate = dateObject.toLocaleDateString('en-US', options);
+    return entryDate;
+  };
+
   return (
     <div>
-      <Navigation headerTitle={headerTitle} />
+      {entry && <Navigation headerTitle={getDate(entry.date)} />}
       <div className={styles.container}>
-        <img className={styles.image} src={src} alt="my picture of the day" />
-        <h1>{entryTitle}</h1>
-        <p className={styles.text}>{text}</p>
+        {entry && (
+          <img
+            className={styles.image}
+            src={entry.imageUrl}
+            alt="my picture of the day"
```

Comment:
Would suggest a more specific name or delete it, if it should be more common :)
'My picture of the day' wouldn't give any hint of what can be seen there. :(